forecasttools/data.py

```diff
@@ -0,0 +1,478 @@
+
+
+def make_nhsn_fitted_forecast_idata(
+    nhsn_dataset_path: str,
+    save_directory: str,
+    file_save_name: str,
+    start_date: str = "2022/08/08",
+    end_date: str = "2023/12/08",
+    forecast_days: int = 28,
+    juris_subset: list[str] = ["TX"],
+    create_save_directory: bool = False,
+    show_plot: bool = True,
+    save_idata: bool = True,
+):
+    """
+    Make an example forecast idata object
+    using a spline regression model on
+    NHSN influenza hospitalization count
+    data.
+    """
+    # check if the save directory exists
+    if not os.path.exists(save_directory):
+        if create_save_directory:
+            os.makedirs(save_directory)
+            print(f"Directory {save_directory} created.")
+        else:
+            raise FileNotFoundError(
+                f"The directory {save_directory} does not exist."
+            )
+    # check if save directory is an actual directory
+    if not os.path.isdir(save_directory):
+        raise NotADirectoryError(
+            f"The path {save_directory} is not a directory."
+        )
+    # check if the save (output) file already exists
+    file_save_path = os.path.join(save_directory, file_save_name)
+    if os.path.exists(file_save_path):
+        raise FileExistsError(f"The file {file_save_path} already exists.")
+    # check that dataset exists
+    if not os.path.exists(nhsn_dataset_path):
+        raise FileNotFoundError(
+            f"The file {nhsn_dataset_path} does not exist."
+        )
+    # check that the loaded CSV has the needed columns
+    df_scan = pl.scan_csv(nhsn_dataset_path)
+    df_cols = df_scan.columns
+    required_cols = ["state", "date", "hosp"]
+    if len(set(required_cols).intersection(set(df_cols))) != len(
+        required_cols
+    ):
+        raise ValueError(
+            f"NHSN dataset missing required columns: {set(required_cols) - set(required_cols).intersection(set(df_cols))}"
+        )
+    # load dataset
+    nhsn = pl.read_csv(nhsn_dataset_path)
+    # check if provided jurisdictions are in NHSN jurisdictions
+    nhsn_juris = list(nhsn["state"].unique().to_numpy())
+    assert len(set(juris_subset).intersection(set(nhsn_juris))) == len(
+        juris_subset
+    ), f"There are jurisdictions present that are not found in the dataset.\nEntered {juris_subset}, Available: {nhsn_juris}"
+    # clean data and organize data, cleaning null values
+    nhsn = nhsn.filter(pl.col("hosp").is_not_null())
+    nhsn = nhsn.filter(pl.col("state").is_in(juris_subset))
+    nhsn = nhsn.with_columns(
+        pl.col("hosp").cast(pl.Int64),
+        pl.col("date").str.strptime(pl.Date, "%Y/%m/%d"),
+    )
+    nhsn = nhsn.filter(
+        (
+            pl.col("date")
+            >= pl.lit(start_date).str.strptime(pl.Date, "%Y-%m-%d")
+        )
+        & (
+            pl.col("date")
+            <= pl.lit(end_date).str.strptime(pl.Date, "%Y-%m-%d")
+        )
+    )
+    nhsn = nhsn.group_by("state").agg([pl.col("hosp")])
```

Comment:
better to chain all these together rather than continually overwriting `nhsn`?

Reply:
No longer immediately relevant given that parent function has been removed.